tools/storage-advisor/src/s3Issues.go

```diff
@@ -345,26 +498,31 @@ func summarizeIssues(results []Finding, checkName string) {
 		})
 		// print issuesWithCounts
 		for _, i := range issueWithCounts {
-			log.Printf("- %d buckets: %s\n", i.Count, i.Issue)
+			fmt.Printf("- %d buckets: %s\n", i.Count, i.Issue)
 		}
 	} else {
-		log.Printf("%s: all good\n", checkName)
+		fmt.Printf("%s: %s\n", bold(checkName), green("%d OK", len(results)))
 	}
-	log.Printf("\n")
+	fmt.Printf("\n")
 }
 
-func runS3Checks(cfg aws.Config, err error) {
-	// Create an S3 service client.
+func runS3Checks(cfg aws.Config, api *Api, identity string) []Bucket {
+	fmt.Printf(green("Running S3 checks\n\n"))
+
 	client := s3.NewFromConfig(cfg)
 
-	// Call the function to list S3 buckets.
 	bucketNames, err := listS3Buckets(client)
 	if err != nil {
-		log.Println("Error:", err)
-		return
+		var ae smithy.APIError
+		if errors.As(err, &ae) {
+			color.Red("Could not list buckets: %s: %s", ae.ErrorCode(), ae.ErrorMessage())
```

Comment:
I believe `fmt.Sprintf("%s: %s", e.ErrorCode(), e.ErrorMessage())` is exactly the way `smithy-go-codegen` implements `smithy.APIError` interface.

Reply:
I'll need to double-check; here I think I did this explicitly because default one looked too verbose.